src/frequenz/dispatch/_managing_actor.py

```diff
@@ -94,58 +131,70 @@ async def run():
         status_receiver = dispatcher.running_status_change.new_receiver()
 
         managing_actor = DispatchManagingActor(
-            actor=my_actor,
+            actor_factory=labda initial_dispatch: MyActor.new_with_dispatch(
+                initial_dispatch, dispatch_updates_channel.new_receiver(),
+            ),
             dispatch_type="EXAMPLE",
             running_status_receiver=status_receiver,
             updates_sender=dispatch_updates_channel.new_sender(),
         )
 
-        await asyncio.gather(dispatcher.start(), managing_actor.start())
+        await run(managing_actor)
     ```
     """
 
     def __init__(
         self,
-        actor: Actor | Set[Actor],
+        actor_factory: Callable[[DispatchUpdate], Actor],
         dispatch_type: str,
         running_status_receiver: Receiver[Dispatch],
         updates_sender: Sender[DispatchUpdate] | None = None,
     ) -> None:
         """Initialize the dispatch handler.
 
         Args:
-            actor: A set of actors or a single actor to manage.
+            actor_factory: A callable that creates an actor with some initial dispatch
+                information.
             dispatch_type: The type of dispatches to handle.
             running_status_receiver: The receiver for dispatch running status changes.
             updates_sender: The sender for dispatch events
         """
         super().__init__()
         self._dispatch_rx = running_status_receiver
-        self._actors: frozenset[Actor] = frozenset(
-            [actor] if isinstance(actor, Actor) else actor
-        )
+        self._actor_factory = actor_factory
+        self._actor: Actor | None = None
```

Comment:
I thought this was going to change to `self._actors: dict[int, Actor]` where the key is int as the dispatch ID, allowing all incoming dispatches to run or `frozenset[int]` allowing only one dispatch for a given set of batteries.

The power setting actor is using int as key.  Is that coming in a separate PR @Marenz?

Reply:
OK, that's a good point but a separate issue. This was a list of actors in the past only because we some use case actor needed to instantiate many actors when a dispatch arrived only because that actor was not designed properly, this is why this PR removes that.

We need to support the use case you mention too, but I would discuss it elsewhere.

Reply:
(this PR might not be merged in a LONG time, while the feature you mention should be implemented soon)